CompuCell3D/core/CompuCell3D/Field3D/Dim3D.cpp

```diff
@@ -22,3 +22,53 @@
 
 #include "Dim3D.h"
 using namespace CompuCell3D;
+
+Dim3D::Dim3D() : Point3D() {}
+
+Dim3D::Dim3D(const short x, const short y, const short z) : Point3D(x, y, z) {}
+
+Dim3D::Dim3D(const Dim3D &dim) : Point3D(dim) {}  
+
+Dim3D &Dim3D::operator=(const Dim3D pt) {
+    x = pt.x;
+    y = pt.y;
+    z = pt.z;
+    return *this;
+}
+
+Dim3D &Dim3D::operator+=(const Dim3D pt) {
```

Comment:
I guess all those functions were in the header file because we were hoping that  compiler might inline them. Perhaps new compiler would optimize it to the point that there is not difference between current and previous codes. Since you have those two branches Could you do a quick benchamrking to see if defining some functions in the header file even matters?

Reply:
Using regression tests, inline functions here on average are 28.5% faster. From a standpoint of readability, I'll just migrate the functions back to the header file and implement them outside the class definitions.